wordpress-mobile/WordPress-Android
# Stats revamp v2 update prompt dialog

This PR adds an alert dialog to inform about **new Insights** if they do not come to Stats through Feature announcement after upgrading to new Jetpack version.  Since we're updating irrespective of their action on Feature announcement.

NOTE:  _Previous implementation using App Preferences to check if cards are updated wouldn't work for users with multiple sites._.  Also, this is a simplified alert dialog with only one action OK, unlike initial designs where assumptions were different.

Fixes #16854 

<img width=320 src="https://user-images.githubusercontent.com/990349/177695806-348f0645-10d5-4d03-9475-1a66798862c5.png" />

To test:

Test 1

- Launch Jetpack app.  
- If you see New Insights Feature announcement dialog tap on Try it now
- Ensure it opens Stats and shows new Insights
- Return to My Site and switch to another site
- Tap on Stats and go to stats
- Verify it shows an update dialog as shown above and upgrades to new Insights

Test 2

- Fresh install Jetpack app
- If you see New Insights Feature announcement dialog tap on Remind me later
- From My Site go to Stats.  If not on Insights tab then switch to it
- Verify it shows an update dialog as shown above and upgrades to new Insights

Test 3
- Try a few other combinations to make sure Insights are updated through both Feature announcement or not.


## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested various scenarios

3. What automated tests I added (or what prevented me from doing so)
Existing unit tests

PR submission checklist:
